plugins/experimental/wasm/wasm_main.cc

```diff
@@ -243,15 +246,25 @@ http_event_handler(TSCont contp, TSEvent event, void *data)
 
   TSMutexUnlock(old_wasm->mutex());
 
-  if (result == 0) {
-    TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
-  } else if (result < 0) {
-    TSHttpTxnReenable(txnp, TS_EVENT_HTTP_ERROR);
+  // check if we have reenable transaction already or not
+  if ((context == nullptr) || (!context->isTxnReenable())) {
+    TSDebug(WASM_DEBUG_TAG, "[%s] no context or not yet reenabled transaction", __FUNCTION__);
+
+    if (result == 0) {
+      TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
```

Comment:
Why does `context->reenable_txn_` not need to get set here?

Reply:
Yes. I should do that. Will make a change.